lib/sort_by_length.rb

```diff
@@ -1,7 +1,20 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2)
+# Space complexity: O(n)
 def sort_by_length(my_sentence)
-  raise NotImplementedError, "Method not implemented"
-end
+  raise ArgumentError unless my_sentence.class == String
+
+  sentence_array = my_sentence.split(" ")
+  my_array = []
+  my_length = sentence_array.length
+  
+  until my_array.length == my_length
+    sentence_array.length.times do |i|
+      shortest = sentence_array.min { |x, y| x.length <=> y.length }
+      my_array << shortest 
+      sentence_array.delete(shortest)
```

Comment:
Note that `.delete` is an O(n) operation since it shifts subsequent elements in the array over by one index.  So this makes your method O(n^3), which is not good.  Instead of deleting an element, I suggest looking at my example sorts and using a swapping strategy instead.